### PR TITLE
arc_to_nurbs function

### PR DESCRIPTION
**Pull request by jvictornreis**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in english)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix:
- [x] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:
Was added a new function to return the necessary parameters to create a NURBS curve equivalent to a given arc

### CODE REVIEW

File: volmdlr/edges.py [hunk 1]
```diff
@@ -2138,6 +2138,54 @@ def _trim_periodic(self, point1: volmdlr.Point3D, point2: volmdlr.Point3D, same_
...
+
+        :param arc: 2D or 3D arc
+        :return: control_points list, knots list, knot_multiplicities list, weights list
+        """
+
+        if arc.angle <= math.pi / 2:
+            number_of_arcs = 1
+        elif arc.angle <= math.pi:
+            number_of_arcs = 2
+        elif arc.angle <= (3 * 2) * math.pi:
```
Comment (GabrielJMS) [inline volmdlr/edges.py]:
```suggestion
        elif arc.angle <= 1.5 * math.pi:
```

File: volmdlr/edges.py [hunk 2]
```diff
@@ -2138,6 +2138,54 @@ def _trim_periodic(self, point1: volmdlr.Point3D, point2: volmdlr.Point3D, same_
...
+        number_of_control_points = 2 * number_of_arcs + 1
+        weight = math.cos(individual_arc_angle / 2)
+        control_points = []
+        weights = []
+        knots = []
+        knot_multiplicities = []
+        knots.insert(0, 0.)
+        knot_multiplicities.insert(0, 3)
+        control_points.insert(0, arc.start)
+        weights.insert(0, 1.)
```
Comment (GabrielJMS) [inline volmdlr/edges.py]:
```suggestion
        control_points = [arc.start]
        weights = [1.0]
        knots = [0.0]
        knot_multiplicities = [3]
```

Comment (GabrielJMS) [reply volmdlr/edges.py]: You can simplify this block by creating the lists directly with the values

File: volmdlr/edges.py [hunk 3]
```diff
@@ -2138,6 +2138,54 @@ def _trim_periodic(self, point1: volmdlr.Point3D, point2: volmdlr.Point3D, same_
...
+
+        for i in range(number_of_arcs):
+            common_control_point = arc.point_at_abscissa((i + 1) * individual_arc_angle * arc.radius)
+            vector_start = (control_points[i * 2] - arc.center).to_vector()
+            vector_end = (common_control_point - arc.center).to_vector()
```
Comment (GabrielJMS) [inline volmdlr/edges.py]: Normally, points subtraction should return a vector, so you don't need to convert it into a vector. Can you verify that, please?

File: tests/edges/test_bsplinecurve.py [hunk 4]
```diff
@@ -17,6 +17,66 @@
...
+        arc = vme.Arc2D.from_3_points(volmdlr.Point2D(0, 0.3), volmdlr.Point2D(1, -0.3), volmdlr.Point2D(2, 2))
+        control_points, knots, knot_multiplicities, weights = vme.BSplineCurve.arc_to_nurbs(arc)
+
+        expected_ctrlpts = [
+            [0.0, 0.3], [0.4578268848009306, -0.4077103355939562],
+            [1.2927631248147025, -0.2922077353431394], [2.127699364828474, -0.17670513509232277],
+            [2.3762039959055192, 0.6287170050840547], [2.6247086269825646, 1.4341391452604322],
+            [2.0000000000000004, 1.9999999999999996]
+        ]
+        expected_knots = [0.0, 0.3333333333333333, 0.6666666666666666, 1.0, 1.0]
```
Comment (GabrielJMS) [inline tests/edges/test_bsplinecurve.py]: To be in line with our definition, it is expected that knots do not have repeating values. For this case, I think it should be rewritten like this:
       expected_knots = [0.0, 0.3333333333333333, 0.6666666666666666, 1.0]
       expected_knot_multiplicities = [3, 2, 2, 5]
 
 The resulting knot-vector would be U = [0.0, 0.0, 0.0, 0.333, 0.333, 0.6666, 0.6666, 1.0, 1.0, 1.0, 1.0, 1.0]
 
 Verify that this is as expected by the algorithm. By the definition the size of the knot vector should be equal to: len(control_points) + degree + 1. As you have 7 control_points, and I imagine the degree is 2, we should expect a knot vector of size 10. So I think maybe the knot 1.0, with multiplicity 2 should not exist for this case? Can you validate this please?
       
image
image

**Review by GabrielJMS — CHANGES_REQUESTED** [review]

File: volmdlr/edges.py (hunk 3 above)
Comment (jvictornreis) [reply volmdlr/edges.py:2174]: I tried it, but didn't work 
WhatsApp Image 2024-03-29 at 12 11 14

**Review by GabrielJMS — APPROVED** [review]

**Pull request merged**